### PR TITLE
feature: add UI & Screen

### PR DESCRIPTION
**Pull request by PIut0**
## 📖 Feature
- Add the Main Menu UI
- Add Game Setting Screen
- Add Earned Coin UI to Game Over Screen
- Add Achievement Screen

## 🔔 Pull Request Type
What changes have been made?

- [x] Add new features
- [ ] Bug fixes
- [ ] Changes that don't affect the code (fix typos, change tab size, rename variables)
- [ ] Refactoring code
- [ ] Adding and modifying comments
- [ ] Documentation modifications
- [ ] Adding tests, refactoring tests
- [ ] Modify build part or package manager
- [ ] Rename a file or folder
- [ ] Delete a file or folder

## 📷 Screen Shot (optional)
|feature|screen shot|
|---|---|
|Main Menu|MainMenu|
|Game Setting Screen|image|
|Game Over|image|
|Achievement|image|

## 🗒️ Notes (optional)

### CODE REVIEW

File: src/engine/GameState.java
Comment (bumsoft) [inline src/engine/GameState.java:1]: The coins earned will be awarded after the game ends in proportion to the score in gameState, so the coinsEarned field is not necessary!
획득한 코인은 gameState의 score에 비례해 게임 종료후 지급할 예정이라 coinsEarned 필드는 없어도 괜찮습니다!

**Review by bumsoft — APPROVED** [review]
LGTM. By the way, will the handling of game settings like the saved isMultiplayer in GameSettingScreen be implemented by the two-player team?
_GameSettingScreen에서 설정된 isMultiplayer 등의 gamesetting 처리는 2인플레이팀에서 구현하게되는 부분인가요?_

**Comment by logpacket** [comment]
> LGTM. By the way, will the handling of game settings like the saved isMultiplayer in GameSettingScreen be implemented by the two-player team? _GameSettingScreen에서 설정된 isMultiplayer 등의 gamesetting 처리는 2인플레이팀에서 구현하게되는 부분인가요?_

Yes. we wil handle that issue.
넵 저희가 구현하게 될 예정입니다.

File: src/engine/DrawManager.java [hunk 1]
```diff
@@ -156,11 +161,12 @@ public void initDrawing(final Screen screen) {
 		backBufferGraphics
 				.fillRect(0, 0, screen.getWidth(), screen.getHeight());
 
+		fontSmallMetrics = backBufferGraphics.getFontMetrics(fontSmall);
 		fontRegularMetrics = backBufferGraphics.getFontMetrics(fontRegular);
 		fontBigMetrics = backBufferGraphics.getFontMetrics(fontBig);
 
 		// drawBorders(screen);
-		// drawGrid(screen);
+		//drawGrid(screen);
```
Comment (logpacket) [inline src/engine/DrawManager.java]: This is an unnecessary change.

**Review by logpacket — COMMENTED** [review]
LGTM. but please remove unnecessary change.

File: src/screen/AchievementScreen.java [hunk 2]
```diff
@@ -13,7 +13,7 @@
  * @author <a href="mailto:RobertoIA1987@gmail.com">Roberto Izquierdo Amo</a>
  * 
  */
-public class HighScoreScreen extends Screen {
+public class AchievementScreen extends Screen {
 
 	/** List of past high scores. */
 	private List<Score> highScores;
```
Comment (fkrdud1125) [inline src/screen/AchievementScreen.java:19]: I think this needs to be changed.

File: src/engine/GameState.java
Comment (PIut0) [reply src/engine/GameState.java:1]: Feel free to modify this as needed by the coin team after the merge...!
해당 내용은 Merge 이후 코인 팀에서 필요하신대로 자유롭게 수정하시면 될 것 같습니다..!

**Review by bumsoft — APPROVED** [review]

**Review by logpacket — APPROVED** [review]

File: src/screen/AchievementScreen.java [hunk 3]
```diff
@@ -69,7 +69,7 @@ protected final void update() {
 	private void draw() {
 		drawManager.initDrawing(this);
 
-		drawManager.drawHighScoreMenu(this);
+		drawManager.drawAchievementMenu(this);
 		drawManager.drawHighScores(this, this.highScores);
```
Comment (fkrdud1125) [inline src/screen/AchievementScreen.java:73]: Is there a reason why you didn't changed the drawHighScores?
혹시 drawHighScores는 따로 변경 안 하신 이유가 있을까요?

**Review by fkrdud1125 — APPROVED** [review]
LGTM

**Review by idealtrue0 — APPROVED** [review]
LGTM
